### DISCUSSION TITLE
How to select multiple AP ?

### DISCUSSION THREAD

**Discussion by arsium** [risinek/esp32-wifi-penetration-tool#30] [Q&A]
Title says all (how to select multiple AP for deauth ?)

**Comment by risinek** [comment]
It's not possible ATM. It would need to be implemented first, and I won't promise anything. Maybe someone else can add support for this.
Attacking multiple APs doesn't add much "research" value, and this project was created mainly as a prove, that attacks on WPA handshakes (including deauth) can be done on ESP32 platform.

**Reply by arsium** [reply]
Ok ok no prob😉

**Reply by risinek** [reply]
To give you some idea how you can achieve deauthing all nearby APs at once (not tested though, just by looking into code)
1. here you can get all scanned APs https://github.com/risinek/esp32-wifi-penetration-tool/blob/c3a1813c4a147a26bb78f6375b2b7fbdbc8f1b26/components/wifi_controller/ap_scanner.c#L41-L43
2. and then you just need to call this method for each AP https://github.com/risinek/esp32-wifi-penetration-tool/blob/c3a1813c4a147a26bb78f6375b2b7fbdbc8f1b26/main/attack_method.c#L38

So basically you can implement the main method `main.c#app_main(void)` like this and it **might** work:
```
wifictl_scan_nearby_aps()
aps = wifictl_get_ap_records()
for(ap from aps) {
 attack_method_broadcast(ap, period);
}
```
but you will probably have to create new handles for `esp_timer_create` for each AP, and take care of initialisation of WiFi etc. The idea is there, but it will definitely need some insight into ESP IDF from your side
